status/condition_set.go

```diff
@@ -237,3 +239,62 @@ func (c ConditionSet) findUnhealthyDependents() []Condition {
 	})
 	return conditions
 }
+
+// unstructuredAdapter is an adapter for the status.Object interface. unstructuredAdapter
+// makes the assumption that status conditions are found on status.conditions path.
+type unstructuredAdapter struct {
+	*unstructured.Unstructured
+}
+
+// FromUnstructured makes the assumption that the status conditions are found on status.conditions
+// path. If they are not found then we return nil.
+func FromUnstructured(u *unstructured.Unstructured) (Object, error) {
+	c, found, err := unstructured.NestedSlice(u.Object, "status", "conditions")
+	if err != nil || !found {
+		return nil, fmt.Errorf("unable to determine status conditions: %w", err)
+	}
+	for _, condition := range c {
```

Comment:
nit: Consider commenting above this with something like, "validating status conditions are `metav1.Condition` types so it's a bit quicker to grok